packages/app/src/files-manager/actions/upload-file.action.ts

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) Mudita sp. z o.o. All rights reserved.
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { DeviceType } from "@mudita/pure"
+import { createAsyncThunk } from "@reduxjs/toolkit"
+import { State } from "App/core/constants"
+import { ReduxRootState } from "App/__deprecated__/renderer/store"
+import {
+  FilesManagerEvent,
+  EligibleFormat,
+  DeviceDirectory,
+} from "App/files-manager/constants"
+import { getPathsRequest } from "App/file-system/requests"
+import { uploadFilesRequest } from "App/files-manager/requests"
+import { getFiles } from "App/files-manager/actions/get-files.action"
+import { setUploadingState } from "App/files-manager/actions/base.action"
+
+export const uploadFile = createAsyncThunk(
+  FilesManagerEvent.UploadFiles,
+  async (_, { getState, dispatch, rejectWithValue }) => {
+    const state = getState() as ReduxRootState
+
+    const filesToUpload = await getPathsRequest({
+      filters: [
+        {
+          name: "Audio",
+          extensions: [
+            EligibleFormat.FLAC,
+            EligibleFormat.MP3,
+            EligibleFormat.WAV,
+          ],
+        },
+      ],
+      properties: ["openFile", "multiSelections"],
+    })
+
+    if (!filesToUpload.ok || !filesToUpload.data) {
+      return rejectWithValue(filesToUpload.error)
+    }
+
+    dispatch(setUploadingState(State.Loading))
+
+    const directory =
+      state.device.deviceType === DeviceType.MuditaHarmony
+        ? DeviceDirectory.Relaxation
+        : DeviceDirectory.Music
+
+    const result = await uploadFilesRequest({
+      directory,
+      paths: filesToUpload.data,
+    })
+
+    if (!result.ok) {
+      return rejectWithValue(result.error)
+    }
+
+    dispatch(setUploadingState(State.Loaded))
+    await dispatch(getFiles(directory))
```

Comment:
Is the order correct? Shouldn't you first wait for the result of `dispatch(getFiles(directory))` and later you should `dispatch(setUploadingState(State.Loaded))`?

Reply:
https://github.com/mudita/mudita-center/pull/914/commits/b57e8f74e6f7294627c128c37a4d33837de9dd8f